modules/phase_field/src/userobjects/PolycrystalCircles.C

```diff
@@ -0,0 +1,189 @@
+  params.addClassDescription(
+      "Polycrystal circles generated from a vector input or read from a file");
+  params.addParam<bool>(
+      "read_from_file", false, "Set to true to read the position and radius vectors from a file rather than inputing them manually");
+  params.addParam<bool>(
+      "columnar_3D", false, "3D microstructure will be columnar in the z-direction?");
+  params.addParam<std::vector<Real> >("x_positions", "x coordinate for each circle center");
+  params.addParam<std::vector<Real> >("y_positions", "y coordinate for each circle center");
+  params.addParam<std::vector<Real> >("z_positions", "z coordinate for each circle center");
+  params.addParam<std::vector<Real> >("radii", "The radius for each circle");
+  params.addParam<FileName>("file_name", "File containing circle centers and radii");
+  params.addParam<unsigned int>("header_length", 0, "Number of lines to skip at top of file");
+
+  return params;
+}
+
+PolycrystalCircles::PolycrystalCircles(const InputParameters & parameters)
+  : PolycrystalUserObjectBase(parameters),
+    _columnar_3D(getParam<bool>("columnar_3D")),
+    _grain_num(0)
+{
+}
+
+void
+PolycrystalCircles::getGrainsBasedOnPoint(const Point & point,
+                                          std::vector<unsigned int> & grains) const
+{
+  unsigned int n_grains = _centerpoints.size();
+  unsigned int cnt = 0;
+  std::vector<bool> is_at_point;
+  is_at_point.resize(n_grains);
+
+  for (unsigned int i = 0; i < n_grains; ++i)
+  {
+    Real d_x = (point(0) - _centerpoints[i](0)) * (point(0) - _centerpoints[i](0));
+    Real d_y = (point(1) - _centerpoints[i](1)) * (point(1) - _centerpoints[i](1));
+    Real d_z = (point(2) - _centerpoints[i](2)) * (point(2) - _centerpoints[i](2));
+    Real distance = 0;
+
+    if (_columnar_3D)
+      distance = std::sqrt(d_x + d_y);
+    else
+      distance = std::sqrt(d_x + d_y + d_z);
+
+    if (distance < _radii[i])
+    {
+      is_at_point[i] = true;
+      ++cnt;
+    }
+    else
+      is_at_point[i] = false;
+  }
+
+  grains.resize(cnt);
+  unsigned int j = 0;
+
+  for (unsigned int i = 0; i < n_grains; ++i)
+  {
+    if (is_at_point[i] == true)
+    {
+      grains[j] = i;
+      ++j;
+    }
+  }
+}
+
+Real
+PolycrystalCircles::getVariableValue(unsigned int op_index, const Point & p) const
+{
+  /*
+  NOTE: This function is identical to the one found in PolycrystalVoronoi.
+  It would be best to pull the function from that file rather than rewriting it
+  here. However, there are two issues that taken together prevent me from doing
+  that:
+
+  1. This file builds on PolycrystalUserObjectBase rather than PolycrystalVoronoi
+  in order to avoid inputting the parameter grain_num, which is not needed here.
+  2. This is a virtual function and must be built either in this file or in the
+  file this builds on.
+
+  How can I get around these issues and inherit the function from PolycrystalVoronoi?
```

Comment:
I've moved away from having a member variable to hold the number of grains. Now there's a virtual you can override to provide this information:

https://github.com/idaholab/moose/blob/devel/modules/phase_field/include/userobjects/PolycrystalUserObjectBase.h#L59